presto-parquet/src/main/java/com/facebook/presto/parquet/reader/ParquetColumnIndexStore.java

```diff
@@ -0,0 +1,160 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.parquet.reader;
+
+import com.facebook.presto.parquet.ParquetDataSource;
+import org.apache.parquet.hadoop.metadata.BlockMetaData;
+import org.apache.parquet.hadoop.metadata.ColumnChunkMetaData;
+import org.apache.parquet.hadoop.metadata.ColumnPath;
+import org.apache.parquet.internal.column.columnindex.ColumnIndex;
+import org.apache.parquet.internal.column.columnindex.OffsetIndex;
+import org.apache.parquet.internal.filter2.columnindex.ColumnIndexStore;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Collections.emptySet;
+
+/**
+ * Internal implementation of {@link ColumnIndexStore}.
+ */
+public class ParquetColumnIndexStore
+        implements ColumnIndexStore
+{
+    private interface IndexStore
+    {
+        ColumnIndex getColumnIndex();
+
+        OffsetIndex getOffsetIndex();
+    }
+
+    private class PageIndexStore
+            implements IndexStore
+    {
+        private final ColumnChunkMetaData columnChunkMetadata;
+        private ColumnIndex columnIndex;
+        private boolean columnIndexRead;
+        private final OffsetIndex offsetIndex;
+
+        PageIndexStore(ColumnChunkMetaData meta)
+        {
+            this.columnChunkMetadata = meta;
+            try {
+                this.offsetIndex = dataSource.readOffsetIndex(meta);
+            }
+            catch (IOException e) {
+                // If the I/O issue still stands it will fail the reading later;
+                // otherwise we fail the filtering only with a missing offset index.
+                throw new MissingOffsetIndexException(meta.getPath());
+            }
+        }
+
+        @Override
+        public ColumnIndex getColumnIndex()
+        {
+            if (!columnIndexRead) {
+                try {
+                    columnIndex = dataSource.readColumnIndex(columnChunkMetadata);
+                }
+                catch (IOException e) {
+                    // If the I/O issue still stands it will fail the reading later;
+                    // otherwise we fail the filtering only with a missing column index.
+                }
+                columnIndexRead = true;
+            }
+            return columnIndex;
+        }
+
+        @Override
+        public OffsetIndex getOffsetIndex()
+        {
+            return offsetIndex;
+        }
+    }
+
+    // Used for columns are not in this parquet file
```

Comment:
we could remove this comment. Or,
`for columns not in this parquet file`

Reply:
We can remove the comments